storm-client/src/jvm/org/apache/storm/utils/NimbusClient.java

```diff
@@ -109,12 +109,13 @@ public static NimbusClient getConfiguredClientAs(Map<String, Object> conf, Strin
         }
 
         List<String> seeds;
-        if(conf.containsKey(Config.NIMBUS_HOST)) {
+        if (conf.containsKey(Config.NIMBUS_HOST)) {
             LOG.warn("Using deprecated config {} for backward compatibility. Please update your storm.yaml so it only has config {}",
-                     Config.NIMBUS_HOST, Config.NIMBUS_SEEDS);
+                    Config.NIMBUS_HOST, Config.NIMBUS_SEEDS);
             seeds = Lists.newArrayList(conf.get(Config.NIMBUS_HOST).toString());
         } else {
-            seeds = (List<String>) conf.get(Config.NIMBUS_SEEDS);
+            Object nimbusSeeds = conf.get(Config.NIMBUS_SEEDS);
+            seeds = nimbusSeeds instanceof String[] ? Lists.newArrayList((String[]) nimbusSeeds) : Lists.newArrayList(nimbusSeeds.toString());
```

Comment:
If this is an administrator configuration setting, can we process the value consistently such that `conf.get(Config.NIMBUS_SEEDS)` always returns a list even if there is only one value. Otherwise, we will have to do this checking at every spot where this value is accessed.

How is this config field validated? Do we allow either a single value or a list?